File: win-prog/ProgWinCS/Edit List and Spin/TextBoxDemo/TextBoxDemo.cs
//------------------------------------------
// TextBoxDemo.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class TextBoxDemo: Form
{
     Label label;

     public static void Main()
     {
          Application.Run(new TextBoxDemo());
     }
     public TextBoxDemo()
     {
          Text = "TextBox Demo";

               // Create text box control.

          TextBox txtbox     = new TextBox();
          txtbox.Parent      = this;
          txtbox.Location    = new Point(Font.Height, Font.Height);
          txtbox.Size        = new Size(ClientSize.Width - 2 * Font.Height,
                                        Font.Height);
          txtbox.Anchor      |= AnchorStyles.Right;
          txtbox.TextChanged += new EventHandler(TextBoxOnTextChanged);

               // Create label control.

          label          = new Label();
          label.Parent   = this;
          label.Location = new Point(Font.Height, 3 * Font.Height);
          label.AutoSize = true;
     }
     void TextBoxOnTextChanged(object obj, EventArgs ea)
     {
          TextBox txtbox = (TextBox) obj;

          label.Text = txtbox.Text;
     }
}

File: win-prog/ProgWinVB/Edit List and Spin/TextBoxDemo/TextBoxDemo.vb
'--------------------------------------------
' TextBoxDemo.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class TextBoxDemo
    Inherits Form

    Private lbl As Label

    Shared Sub Main()
        Application.Run(New TextBoxDemo())
    End Sub

    Sub New()
        Text = "TextBox Demo"

        ' Create text box control.
        Dim txtbox As New TextBox()
        txtbox.Parent = Me
        txtbox.Location = New Point(Font.Height, Font.Height)
        txtbox.Size = New Size(ClientSize.Width - 2 * Font.Height, _
                               Font.Height)
        txtbox.Anchor = txtbox.Anchor Or AnchorStyles.Right
        AddHandler txtbox.TextChanged, AddressOf TextBoxOnTextChanged

        ' Create label control.
        lbl = New Label()
        lbl.Parent = Me
        lbl.Location = New Point(Font.Height, 3 * Font.Height)
        lbl.AutoSize = True
    End Sub

    Private Sub TextBoxOnTextChanged(ByVal obj As Object, _
                                     ByVal ea As EventArgs)
        Dim txtbox As TextBox = DirectCast(obj, TextBox)
        lbl.Text = txtbox.Text
    End Sub
End Class
